projects/dotcms-field-elements/src/components/dot-textfield/dot-texfield.e2e.ts

```diff
@@ -6,135 +6,282 @@ describe('dot-textfield', () => {
+    const getErrorMessage = async () => await page.find('.dot-field__error-message');
+
+    describe('@Props', () => {
+        beforeEach(async () => {
+            page = await newE2EPage({
+                html: `<dot-textfield></dot-textfield>`
+            });
+
+            element = await page.find('dot-textfield');
+            input = await page.find('input');
         });
 
-        spyStatusChangeEvent = await page.spyOnEvent('statusChange');
-        spyValueChange = await page.spyOnEvent('valueChange');
-        element = await page.find('dot-textfield');
-        input = await page.find('input');
-    });
+        describe('value', () => {
+            it('should set value correctly', async () => {
+                element.setProperty('value', 'test');
+                await page.waitForChanges();
 
-    it('should render', () => {
-        // tslint:disable-next-line:max-line-length
-        const tagsRenderExpected = `<div class=\"dot-field__label\"><label for=\"dot-fullName\">Name:</label><span class=\"dot-field__required-mark\">*</span></div><input id=\"dot-fullName\" placeholder=\"Enter Name\" required=\"\" type=\"text\"><span class=\"dot-field__hint\">this is a hint</span>`;
-        expect(element.innerHTML).toBe(tagsRenderExpected);
-    });
+                expect(await input.getProperty('value')).toBe('test');
+            });
+            it('should render and not break when is a unexpected value', async () => {
+                element.setProperty('value', { test: true });
+                await page.waitForChanges();
 
-    it('should set type', async () => {
-        element.setProperty('type', 'email');
-        await page.waitForChanges();
-        expect(await input.getProperty('type')).toBe('email');
-    });
+                expect(await input.getProperty('value')).toBe('[object Object]');
+            });
+        });
 
-    it('should show Regex validation message', async () => {
-        await input.press('@');
-        await page.waitForChanges();
-        const errorMessage = await page.find('.dot-field__error-message');
-        expect(errorMessage.innerHTML).toBe('Invalid Name');
-    });
+        describe('name', () => {
+            it('should render with valid name', async () => {
+                element.setProperty('name', 'text01');
+                await page.waitForChanges();
+                expect(await input.getProperty('id')).toBe('dot-text01');
+            });
 
-    it('should load as pristine and untouched', () => {
-        expect(element.classList.contains('dot-pristine')).toBe(true);
-        expect(element.classList.contains('dot-untouched')).toBe(true);
-    });
+            it('should set name prop in dot-label', async () => {
+                element.setProperty('name', 'text01');
+                await page.waitForChanges();
+                const label = await getLabel();
+                expect(await label.getProperty('name')).toBe('text01');
```

Comment:
`getProperty` or `getAttribute`? what shuold we use?

Reply:
`getProperty ` for custom elements.